File: python/bob/bob.py

class Bob:

	def __init__(self):
		pass

	def hey(self, message):
		if message is None or len(trim(message)) == 0:
			return 'Fine. Be that way!'
		if message.isupper():
			return 'Woah, chill out!'
		if message.endswith('?'):
			return 'Sure.'
		return 'Whatever.'

File: clojure/bob/bob.clj
(ns bob)
(use 'clojure.string)
(defn response-for
  "Bob responds here."
  [msg]
  (if (or (nil? msg) (blank? msg))
    "Fine. Be that way!"
    ; If all alphabetic characters are upper case...
    (if (and (= (upper-case msg) msg) (not= (lower-case msg) msg))
        "Woah, chill out!"
    (if (= \? (last msg))
      "Sure."
      "Whatever."
      ))))
